### ISSUE TITLE
lib not found: shiboken2.abi3.dll dependency

### ISSUE THREAD

**Issue by fredrikaverpil** [mherrmann/fbs#119]
OS: Windows 10, May 2019 update
Python: Minconda 3.7.3 (unsupported, I know)
PySide2: 5.12.3
fbs: 0.8.2

When freezing with `--debug` I see this flying by in the log:

```
WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\QtWidgets.pyd
WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\QtGui.pyd
WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\pyside2.abi3.dll
WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\QtCore.pyd
WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\QtNetwork.pyd
WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\pyside2.abi3.dll
```

- I can locate `shiboken2.abi3.dll` here: `./venv/Lib/site-packages/shiboken2/pyside2.abi3.dll`.
- There is a closed issue on this for pyinstaller: https://github.com/pyinstaller/pyinstaller/issues/4044
- There is a bug in the PySide2 tracker but it seems they consider it fixed in PySide2 5.12.3, which I am using: https://bugreports.qt.io/browse/PYSIDE-1002
- In my venv, I try to `import shiboken2` and that works without issues.

Instead of dealing with pyinstaller hooks, I just import the packages which are not automatically detected in my fbs app. This has worked fine so far. However, it is not sufficient to `import shiboken2` in this case. @mherrmann do you have any ideas on how this could be solved?

**Comment by mherrmann** [comment]
Hi Fredrik, does the problem also occur with Python 3.6?

**Comment by mherrmann** [comment]
(Vanilla Python 3.6)

**Comment by fredrikaverpil** [comment]
Using vanilla Python 3.6.8, I get the same warnings.

**Comment by mherrmann** [comment]
Thank you. If I understand correctly, your application works fine and you're just worried about the warning? If this is the case: PyInstaller 3.4 prints several wrong WARNINGS and fbs has a workaround to suppress them (see [this code comment](https://github.com/mherrmann/fbs/blob/f1d84da8d017358db08edfd21c03d2990e39b37e/fbs/freeze/__init__.py#L19-L21)). However, when the `--debug` flag is given, fbs's workaround is disabled.

In short, I think this is a false warning by PyInstaller, with no adverse effects. If this does cause problems in your environment, let me know by re-opening this issue. And thank you again for taking the time to report.

**Comment by fredrikaverpil** [comment]
> If I understand correctly, your application works fine and you're just worried about the warning? 

Correct.

**Comment by mherrmann** [comment]
Okay, thanks. Then I'd leave it in fbs as-is for now. As I said, `fbs freeze` filters out this warning (and others like it). As long as it doesn't have any adverse effects, I think it shouldn't be a problem. (Though admittedly not nice.)

**Comment by fredrikaverpil** [comment]
Just now I noticed that these warnings appear when freezing without `--debug` too.

**Comment by mherrmann** [comment]
That would be strange. The output you pasted above are `WARNING`s. And fbs sets the log level to `ERROR`. ([Code](https://github.com/mherrmann/fbs/blob/f1d84da8d017358db08edfd21c03d2990e39b37e/fbs/freeze/__init__.py#L22))

**Comment by fredrikaverpil** [comment]
```
0 INFO: PyInstaller: 3.4
91 INFO: Python: 3.7.3
93 INFO: Platform: Windows-10-10.0.18362-SP0
1048 INFO: wrote C:\Users\fredrik\code\repos\skalman\target\PyInstaller_launch\launch.spec
1054 INFO: UPX is not available.
1056 INFO: Extending PYTHONPATH with paths
['C:\\Users\\fredrik\\code\\repos\\skalman\\src\\main\\python',
 'C:\\Users\\fredrik\\code\\repos\\skalman\\target\\PyInstaller_launch']
1058 INFO: checking Analysis
1060 INFO: Building Analysis because Analysis-00.toc is non existent
1060 INFO: Initializing module dependency graph...
1088 INFO: Initializing module graph hooks...
1091 INFO: Analyzing base_library.zip ...
5537 INFO: running Analysis Analysis-00.toc
5540 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\users\fredrik\code\repos\skalman\venv\scripts\python.exe
5980 INFO: Caching module hooks...
5987 INFO: Analyzing C:\Users\fredrik\code\repos\skalman\src\main\python\skalman\applauncher\launch.py
6001 INFO: Processing pre-find module path hook   site
6013 INFO: site: retargeting to fake-dir 'c:\\users\\fredrik\\code\\repos\\skalman\\venv\\lib\\site-packages\\PyInstaller\\fake-modules'
8779 INFO: Processing pre-find module path hook   distutils
13255 INFO: Processing pre-safe import module hook   win32com
17056 INFO: Processing pre-safe import module hook   six.moves
17947 INFO: Loading module hooks...
17948 INFO: Loading module hook "hook-distutils.py"...
17950 INFO: Loading module hook "hook-encodings.py"...
18090 INFO: Loading module hook "hook-jinja2.py"...
18112 INFO: Loading module hook "hook-pkg_resources.py"...
18983 INFO: Loading module hook "hook-psycopg2.py"...
18984 WARNING: Hidden import "mx.DateTime" not found!
18985 INFO: Loading module hook "hook-pydoc.py"...
18987 INFO: Loading module hook "hook-PySide2.py"...
18991 WARNING: Hidden import "numpy.core.multiarray" not found!
18991 INFO: Loading module hook "hook-PySide2.QtNetwork.py"...
19222 INFO: Loading module hook "hook-pythoncom.py"...
19654 INFO: Loading module hook "hook-pytz.py"...
19776 INFO: Loading module hook "hook-pywintypes.py"...
20198 INFO: Loading module hook "hook-sqlalchemy.py"...
20527 INFO:   Found 6 sqlalchemy hidden imports
20529 WARNING: Hidden import "MySQLdb" not found!
20552 WARNING: Hidden import "sqlalchemy.orm.state", "sqlalchemy.orm.strategies" not found!
20553 WARNING: Hidden import "sqlalchemy.sql.functions.func" not found!
20584 INFO: Excluding import 'sqlalchemy.testing'
20589 INFO:   Removing import of sqlalchemy.testing.util from module sqlalchemy.orm.util
20590 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing
20591 INFO:   Removing import of sqlalchemy.testing.util from module sqlalchemy.testing
20592 INFO:   Removing import of sqlalchemy.testing.engines from module sqlalchemy.testing
20592 INFO:   Removing import of sqlalchemy.testing.config from module sqlalchemy.testing
20593 INFO:   Removing import of sqlalchemy.testing.assertsql from module sqlalchemy.testing
20594 INFO:   Removing import of sqlalchemy.testing.warnings from module sqlalchemy.testing
20621 INFO:   Removing import of sqlalchemy.testing.mock from module sqlalchemy.testing
20622 INFO:   Removing import of sqlalchemy.testing.exclusions from module sqlalchemy.testing
20622 INFO:   Removing import of sqlalchemy.testing.assertions from module sqlalchemy.testing
20622 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.config
20624 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.mock
20624 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.assertions
20626 INFO:   Removing import of sqlalchemy.testing.util from module sqlalchemy.testing.assertions
20626 INFO:   Removing import of sqlalchemy.testing.config from module sqlalchemy.testing.assertions
20627 INFO:   Removing import of sqlalchemy.testing.assertsql from module sqlalchemy.testing.assertions
20627 INFO:   Removing import of sqlalchemy.testing.mock from module sqlalchemy.testing.assertions
20627 INFO:   Removing import of sqlalchemy.testing.exclusions from module sqlalchemy.testing.assertions
20629 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.assertsql
20658 INFO:   Removing import of sqlalchemy.testing.config from module sqlalchemy.testing.util
20659 INFO:   Removing import of sqlalchemy.testing.engines from module sqlalchemy.testing.util
20660 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.util
20660 INFO:   Removing import of sqlalchemy.testing.assertions from module sqlalchemy.testing.engines
20661 INFO:   Removing import of sqlalchemy.testing.util from module sqlalchemy.testing.engines
20662 INFO:   Removing import of sqlalchemy.testing.config from module sqlalchemy.testing.engines
20693 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.engines
20695 INFO:   Removing import of sqlalchemy.testing.config from module sqlalchemy.testing.exclusions
20727 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.exclusions
20729 INFO:   Removing import of sqlalchemy.testing.assertions from module sqlalchemy.testing.warnings
20730 INFO:   Removing import of sqlalchemy.testing from module sqlalchemy.testing.warnings
20732 INFO: Loading module hook "hook-sqlite3.py"...
20882 INFO: Loading module hook "hook-sysconfig.py"...
20885 INFO: Loading module hook "hook-win32com.py"...
20981 INFO: Loading module hook "hook-xml.dom.domreg.py"...
20982 INFO: Loading module hook "hook-xml.etree.cElementTree.py"...
20983 INFO: Loading module hook "hook-xml.py"...
20985 INFO: Loading module hook "hook-PySide2.QtCore.py"...
21277 INFO: Looking for ctypes DLLs
21313 INFO: Analyzing run-time hooks ...
21330 INFO: Including run-time hook 'pyi_rth_pkgres.py'
21354 INFO: Including run-time hook 'pyi_rth_win32comgenpy.py'
21357 INFO: Including run-time hook 'pyi_rth_multiprocessing.py'
21361 INFO: Including run-time hook 'pyi_rth_qt5plugins.py'
21382 INFO: Looking for dynamic libraries
21854 WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\QtCore.pyd
22220 WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\QtNetwork.pyd
23063 WARNING: lib not found: shiboken2.abi3.dll dependency of c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PySide2\pyside2.abi3.dll
23364 INFO: Looking for eggs
23364 INFO: Using Python library C:\Users\fredrik\Miniconda3\python37.dll
23365 INFO: Found binding redirects:
[]
23388 INFO: Warnings written to C:\Users\fredrik\code\repos\skalman\target\PyInstaller_launch\launch\warn-launch.txt
23600 INFO: Graph cross-reference written to C:\Users\fredrik\code\repos\skalman\target\PyInstaller_launch\launch\xref-launch.html
23705 INFO: checking PYZ
23706 INFO: Building PYZ because PYZ-00.toc is non existent
23707 INFO: Building PYZ (ZlibArchive) C:\Users\fredrik\code\repos\skalman\target\PyInstaller_launch\launch\PYZ-00.pyz
26220 INFO: Building PYZ (ZlibArchive) C:\Users\fredrik\code\repos\skalman\target\PyInstaller_launch\launch\PYZ-00.pyz completed successfully.
26263 INFO: checking PKG
26264 INFO: Building PKG because PKG-00.toc is non existent
26266 INFO: Building PKG (CArchive) PKG-00.pkg
26314 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
26317 INFO: Bootloader c:\users\fredrik\code\repos\skalman\venv\lib\site-packages\PyInstaller\bootloader\Windows-64bit\run.exe
26318 INFO: checking EXE
26318 INFO: Building EXE because EXE-00.toc is non existent
26319 INFO: Building EXE from EXE-00.toc
26319 INFO: Appending archive to EXE C:\Users\fredrik\code\repos\skalman\target\PyInstaller_launch\launch\launch.exe
26440 INFO: Building EXE from EXE-00.toc completed successfully.
26453 INFO: checking COLLECT
26474 INFO: Building COLLECT because COLLECT-00.toc is non existent
26475 INFO: Building COLLECT COLLECT-00.toc
Aborted by user request.
```

**Comment by mherrmann** [comment]
Probably. `fbs freeze` typically produces no output at all (beyond a message how the frozen app can be run).

**Comment by fredrikaverpil** [comment]
Edited my post above. It's probably my `_freeze_launch` function which calls pyinstaller via `subprocess`.

**Comment by mherrmann** [comment]
Yeah, I'm pretty confident the `subprocess` call to PyInstaller is what produces all this output.

**Comment by fredrikaverpil** [comment]
Yes, that's it. Sorry for the noise!

**Comment by Kalyan95** [comment]
For the missing library libshiboken2.dll. (libshiboken2.dylib on MacOS and Linux)

`pip install shiboken2 `

and then in site-packages/ shiboken2 you can find the missing libshiboken2.dll which you can copy to site-packages/Pyside2 and it works.

works the same way on MacOS and Linux as well (Libshibkoken2.dylib in those cases.)

Hope this helps someone along the way.

**Comment by Doev** [comment]
> 
> 
> For the missing library libshiboken2.dll. (libshiboken2.dylib on MacOS and Linux)
> 
> `pip install shiboken2 `
> 
> and then in site-packages/ shiboken2 you can find the missing libshiboken2.dll which you can copy to site-packages/Pyside2 and it works.
> 
> works the same way on MacOS and Linux as well (Libshibkoken2.dylib in those cases.)
> 
> Hope this helps someone along the way.

Hello Kalyan95,

in my case shiboken2 is already installed and libshiboken2.dll is already in the directory site-packages/shiboken2. I get this error too.
